### PR TITLE
Add a new rule that detects area/scope wrapped under square brackets instead of "foo: bar" style

### PR DESCRIPTION
**Pull request by realmarv**

### CODE REVIEW

File: commitlint.config.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ module.exports = {
         'trailing-whitespace': [RuleStatus.Error, 'always'],
         'prefer-slash-over-backslash': [RuleStatus.Error, 'always'],
         'type-space-before-paren': [RuleStatus.Error, 'always'],
+        'reject-square-bracket-style': [RuleStatus.Error, 'always'],
```
Comment (knocte) [inline commitlint.config.ts:84]: @realmarv to follow same naming style as commitlint (e.g. "type-empty"), let's rename this to `type-with-square-brackets`

File: commitlint.config.ts [hunk 2]
```diff
@@ -168,6 +168,17 @@ module.exports = {
...
                 },
 
+                'reject-square-bracket-style': ({header}: {header:any}) => {
+                    let headerStr = convertAnyToString(header, "header");
+
+                    let offence = headerStr.match(`^\\[.*\\]`) !== null
+
+                    return [
+                        !offence,
+                        `Please use "area/scope: title" style instead of wrapping area/scope under square brakets in your commit message header`
```
Comment (knocte) [inline commitlint.config.ts:178]: @realmarv it should suggest area/scope or area(scope)

File: commitlint.config.ts [hunk 3]
```diff
@@ -168,14 +168,14 @@ module.exports = {
                     ];
                 },
 
-                'reject-square-bracket-style': ({header}: {header:any}) => {
+                'type-with-square-brackets': ({header}: {header:any}) => {
```
Comment (knocte) [inline commitlint.config.ts:171]: @realmarv but then now this rule needs to be moved down

File: commitlint.config.ts [hunk 4]
```diff
@@ -168,14 +168,14 @@ module.exports = {
...
                     let headerStr = convertAnyToString(header, "header");
 
                     let offence = headerStr.match(`^\\[.*\\]`) !== null
 
                     return [
                         !offence,
-                        `Please use "area/scope: title" style instead of wrapping area/scope under square brakets in your commit message header`
+                        `Please use "area/scope: title" or "area(scope): title" style instead of wrapping area/scope under square brakets in your commit message header`
```
Comment (knocte) [inline commitlint.config.ts:178]: @realmarv typo "brakets" -> "brackets"

File: commitlint.config.ts [hunk 5]
```diff
@@ -197,6 +186,17 @@ module.exports = {
                     ];
                 },
 
+                'type-with-square-brackets': ({header}: {header:any}) => {
```
Comment (knocte) [inline commitlint.config.ts:189]: @realmarv and let's reorder tests too

File: commitlint.config.ts (hunk 3 above)
Comment (realmarv) [reply commitlint.config.ts]: Done

File: commitlint.config.ts (hunk 2 above)
Comment (realmarv) [reply commitlint.config.ts]: Done

File: commitlint.config.ts (hunk 1 above)
Comment (realmarv) [reply commitlint.config.ts]: Done

File: commitlintplugins.test.ts [hunk 6]
```diff
@@ -304,3 +304,17 @@ test('type-space-before-paren3', () => {
...
+test('type-with-square-brackets1', () => {
+    let commitMsgWithSquareBracketStyle = "[foo] this is a title";
+    let rejectSquareBracketStyle1 = runCommitLintOnMsg(commitMsgWithSquareBracketStyle);
+    expect(rejectSquareBracketStyle1.status).not.toBe(0);
+});
+
+
+test('type-with-square-brackets2', () => {
+    let commitMsgWithoutSquareBracketStyle = "foo: this is a title";
+    let rejectSquareBracketStyle2 = runCommitLintOnMsg(commitMsgWithoutSquareBracketStyle);
```
Comment (knocte) [inline commitlintplugins.test.ts:318]: @realmarv renaming a rule has many side-effects sorry, these vars need to be renamed too

File: commitlint.config.ts [hunk 7]
```diff
@@ -186,6 +186,17 @@ module.exports = {
...
+                    let headerStr = convertAnyToString(header, "header");
+
+                    let offence = headerStr.match(`^\\[.*\\]`) !== null
+
+                    return [
+                        !offence,
+                        `Please use "area/scope: title" or "area(scope): title" style instead of wrapping area/scope under square brackets in your commit message header`
```
Comment (knocte) [inline commitlint.config.ts:196]: should be "area/scope: subject"

Comment (knocte) [reply commitlint.config.ts:196]: and should be "commit message title"

Comment (realmarv) [reply commitlint.config.ts]: Done

File: commitlintplugins.test.ts (hunk 6 above)
Comment (realmarv) [reply commitlintplugins.test.ts]: Done

File: commitlint.config.ts (hunk 5 above)
Comment (realmarv) [reply commitlint.config.ts:189]: Done

**Pull request merged**